control_panel_api/views.py

```diff
@@ -227,6 +233,25 @@ def perform_create(self, serializer):
 
         instance.create_users3bucket(user=self.request.user)
 
+    @detail_route()
+    def access_logs(self, request, pk=None):
```

Comment:
Isn't this a `list_route`? As far as I can tell `pk` is not used.
I've only just started to have a look at your PR, so apologies if I'm missing something.

Reply:
`detail_route` applies to the s3bucket so you get the {id} part and object, this is just a sub route